### DISCUSSION TITLE
Does a nested write inside an interactive transaction collapse the transactions?

### DISCUSSION THREAD

**Discussion by aniravi24** [prisma/prisma#15213] [Q&A] [answered]
If you have an interactive transaction that contains a nested write, will that cause an issue since nested writes are done with transactions?

For example
```
await prisma.$transaction(
  async (prisma) => {
    // Code running in a transaction...
    await prisma.post.update({
      where: { id: 42 },
      data: {
        author: {
          connect: { email: 'alice@prisma.io' },
        },
      },
    })
  }
)
```

Does the nested write inside the transaction callback try to make another transaction nested inside of the interactive transaction? Or does the prisma client know to properly use the callback's transaction? Databases like postgres don't support nested transactions so it might cause silent failures if it doesn't work as expected.

Initially mentioned [here](https://github.com/prisma/prisma/issues/8664#issuecomment-1175594282)

**Comment by nikolasburk** [comment, marked as answer]
Hey there 👋 thanks for raising this! 

> Does the nested write inside the transaction callback try to make another transaction nested inside of the interactive transaction? 

No.

> Or does the prisma client know to properly use the callback's transaction?

Yes, Prisma Client will ensure that only one transactions is being opened, even if you include nested writes in an interactive transaction.

**Reply by aniravi24** [reply]
Awesome, thank you!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
